assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -717,81 +758,82 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
   // To calculate the height of that missing part, that is:
   // LINE_HEIGHT*scaleFactor/2 - MAX_ENDPOINT_HEIGHT*scaleFactor/2 + (hasEmphasis ? EMPHASIS_PADDING_TOP * scaleFactor : 0)
 
-  // offset is parent container height minus all the stuff below the very top of the line diagram
+  // So finally, the vertical viewBoxOffset is parent container height minus
+  // all the stuff below the very top of the line diagram
   const viewBoxOffset = 
-    height
-    - (LINE_HEIGHT * scaleFactor) / 2
-    - (MAX_ENDPOINT_HEIGHT * scaleFactor) / 2
+    originalHeight
+    - LINE_HEIGHT * scaleFactor / 2
+    - MAX_ENDPOINT_HEIGHT * scaleFactor / 2
     + (hasEmphasis ? EMPHASIS_PADDING_TOP * scaleFactor : 0)
 
   return (
-    <svg
-      // viewBoxOffset will always be > 0 by the time it's visible, but the console will
-      // still log an error if it's a negative number when it's not-yet-visible
-      viewBox={`0 ${-viewBoxOffset} 904 ${height + getEmphasisHeight(scaleFactor)}`}
-      transform={`translate(${translateX})`}
-      visibility={isDone ? "visible" : "hidden"}
-    >
-      <g transform={`translate(${L * scaleFactor} 0)`}>
-        <g id="line-map" transform={`scale(${scaleFactor})`}>
-          {effect !== "station_closure" && (
-            <EffectBackgroundComponent
-              effectRegionSlotIndexRange={props.effect_region_slot_index_range}
-              effect={effect}
+    <div style={{width: "100%", height: "100%"}} id="diagram-container" ref={ref}>
```

Comment:
Since we can have 2 diagrams on the page at once—currently only in the Screenplay "exploded flex zone", I think—this element's `id` won't always be unique. Would it be possible to change it to a `class` instead?

Same for the other sub-elements with `id`s.
But if it's a huge pain to change, should be ok to leave it as-is.

Reply:
Turns out the id wasn't being used once other things were re-factored. 100% correct though, this wasn't a good uid